pageserver/src/tenant_mgr.rs

```diff
@@ -374,15 +373,7 @@ pub fn get_local_timeline_with_load(
     tenant_id: ZTenantId,
     timeline_id: ZTimelineId,
 ) -> anyhow::Result<Arc<Timeline>> {
-    let repository = get_repository_for_tenant(tenant_id)?;
```

Comment:
This was the main function that kept our state intact despite some logic caps, either initializing the timeline's initial logical size or eagerly failing if some layer was corrupt.

Now this is not true anymore and we're calculating the size lazily, which means we **do not check if the layers are corrupt during timeline load**(!).
I think it's bad, but loading every layer to check if those are corrupt also seems odd to me. IMO, we should add checksum verification around here.